### PR TITLE
feat: recovery queue

### LINKED ISSUE #2763: [Recovery] Transaction queue - group skipped transactions
## What is the feature about

Adding "pending recovery" section of the transaction queue.

Note: >1 proposal can exist.

## The list of requirements

- [x] Add new "pending recovery" section to the transaction queue
- [x] Parse proposals and display them in a transaction details-esque component
  - [x] Display to-be-added owner(s)
  - [x] Populate as many detail fields as possible from the module, e.g. `txNonce`, etc.
  - [x] Create proposal stepper
  - [x] Add execute/cancel button dependent on connected wallet
- [x] Group cancellation with proposal

## Designs/sketches

image

image

image

image

## Links

- [Feature spec.](https://www.notion.so/safe-global/Recovery-feature-spec-d873c33a27c24cfabc91c85dbcf3ea35)
- [Figma](https://www.figma.com/file/pO43pjVwR34Uy1ihMsY3OM/Generic-recovery?type=design&node-id=456-15322&mode=design&t=Lo3nDvw622nW8RA7-0)

### PR DESCRIPTION
**Pull request by iamacook**
## What it solves

Resolves #2763

## How this PR fixes it

This implements a new "Pending recovery" section to the transaction queue which displays all recovery attempts from connected recovery-specific Delay Modifiers, as well as associated adjustments:

- Recovery queue with decoded overview
- Global state adjustments (`executor` and `isMalicious` flags have been added, as well as all timestamps converted to ms)
- Associated unit tests

Note: text was tweaked and does not necessarily match the design. It is also not possible to skip recovery proposals yet. As such, queued recovery attempts and their "skip" transactions are not included in this PR.

## How to test it

1. Propose a recovery attempt
2. Observe recovery proposal in queue, displaying cooldown, expiration and option to execute

## Screenshots

image

Old naming/colouring:
image

image

image

image

## Checklist
* [ ] I've tested the branch on mobile 📱
* [ ] I've documented how it affects the analytics (if at all) 📊
* [x] I've written a unit/e2e test for it (if applicable) 🧑‍💻

### CODE REVIEW

File: src/hooks/useTimestamp.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { useState, useEffect } from 'react'
+
+export function useTimestamp(interval = 1_000): number {
```
Comment (katspaugh) [inline src/hooks/useTimestamp.ts:3]: I would call this hook `useClock` or `useIntervalTimestamp`. `useTimestamp` to me reads as a one-time timestamp.

File: src/components/recovery/RecoveryInfo/index.tsx [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { SvgIcon, Tooltip } from '@mui/material'
+import type { ReactElement } from 'react'
+
+import WarningIcon from '@/public/images/notifications/warning.svg'
+
+export const RecoveryInfo = (): ReactElement => {
+  return (
+    <Tooltip title="This recovery attempt was not created via the interface" placement="top" arrow>
```
Comment (katspaugh) [inline src/components/recovery/RecoveryInfo/index.tsx:8]: > This recovery attempt was not created via the interface

That it wasn't created via the interface is a poor way to say it's a fraudulent transaction.

Design-wise, I wouldn't even display them as recovery attempts. Too easy to confuse with a legit recovery.

Screenshot 2023-11-21 at 11 09 13

File: src/hooks/useTimestamp.ts (hunk 1 above)
Comment (iamacook) [reply src/hooks/useTimestamp.ts]: I've renamed it to `useClock` in 6896ea6497e6318c8ffb226c27737f612c12f456.

File: src/components/recovery/RecoveryInfo/index.tsx (hunk 2 above)
Comment (iamacook) [reply src/components/recovery/RecoveryInfo/index.tsx]: I've renamed the transaction type of "Malicious transaction" (coloured in `error.main`) and changed the tooltip to "Suspicious activity" in 6896ea6497e6318c8ffb226c27737f612c12f456.

image

**Review by katspaugh — APPROVED** [review]

**Pull request merged**